pkg/manager/member/tidb_discovery_manager.go

```diff
@@ -45,29 +47,48 @@ func NewTidbDiscoveryManager(deps *controller.Dependencies) TidbDiscoveryManager
 	return &realTidbDiscoveryManager{deps: deps}
 }
 
-func (m *realTidbDiscoveryManager) Reconcile(tc *v1alpha1.TidbCluster) error {
+func (m *realTidbDiscoveryManager) Reconcile(obj runtime.Object) error {
+	metaObj, ok := obj.(metav1.Object)
+	if !ok {
+		return fmt.Errorf("%T is not a metav1.Object", obj)
+	}
+
+	meta, _, err := getDiscoveryMeta(metaObj, controller.DiscoveryMemberName)
+	if err != nil {
+		return fmt.Errorf("error generating discovery meta: %v", err)
+	}
 
-	// If PD is not specified return
-	if tc.Spec.PD == nil {
-		return nil
+	var (
+		clusterPolicyRule rbacv1.PolicyRule
+	)
+	switch cluster := obj.(type) {
+	case *v1alpha1.TidbCluster:
+		// If PD is not specified return
+		if cluster.Spec.PD == nil {
+			return nil
+		}
+		clusterPolicyRule = rbacv1.PolicyRule{
+			APIGroups:     []string{v1alpha1.GroupName},
+			Resources:     []string{v1alpha1.TiDBClusterName},
+			ResourceNames: []string{metaObj.GetName()},
+			Verbs:         []string{"get"},
+		}
+	case *v1alpha1.DMCluster:
+		clusterPolicyRule = rbacv1.PolicyRule{
+			APIGroups:     []string{v1alpha1.GroupName},
+			Resources:     []string{v1alpha1.DMClusterName},
+			ResourceNames: []string{metaObj.GetName()},
+			Verbs:         []string{"get"},
+		}
+	default:
+		return fmt.Errorf("unsupported type %T for discovery", obj)
```

Comment:
Sorry that I did not make it clear in my last comment. I think no need to return error here, because return error will cause the CR to be requeued and synced again and again, however, with this error, it's useless to resync.

Reply:
changed to log a warn in https://github.com/pingcap/tidb-operator/pull/3914/commits/65fa3ec037fb806cd2940fc9a4b496b42f9c34af.